File: openapi-generator/dart-jaguar/lib/model/document_type.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

import 'package:openapi/model/document_field_type.dart';
part 'document_type.jser.dart';

class DocumentType {
   /* The field types determine the names of fields for this document type. They also determine the types of values a document submitted for this document type can have. */
  @Alias('fieldTypes')
  final List<List> fieldTypes;
   /* Name of the document type. */
  @Alias('name')
  final String name;
   /* ID for the organization this document belongs to. */
  @Alias('orgId')
  final int orgId;
   /* Universally unique identifier for the document type. Can be passed in as a documentTypeUuid when creating a document for this document type. */
  @Alias('uuid')
  final String uuid;
  

  DocumentType(
      

{
     this.fieldTypes = const [],  
    
     this.name = null,  
     this.orgId = null,  
     this.uuid = null 
    }
  );

  @override
  String toString() {
    return 'DocumentType[fieldTypes=$fieldTypes, name=$name, orgId=$orgId, uuid=$uuid, ]';
  }
}

@GenSerializer(nullableFields: true)
class DocumentTypeSerializer extends Serializer<DocumentType> with _$DocumentTypeSerializer {

}

File: openapi-generator/php-ze-ph/src/App/DTO/DocumentType.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 */
class DocumentType
{
    /**
     * The field types determine the names of fields for this document type. They also determine the types of values a document submitted for this document type can have.
     * @DTA\Data(field="fieldTypes", nullable=true)
     * TODO check validator and strategy are correct and can handle container item type
     * @DTA\Strategy(name="ObjectArray", options={"type":\App\DTO\DocumentFieldType[]::class})
     * @DTA\Validator(name="Collection", options={"validators":{
     *     {"name":"Dictionary", "options":{"type":\App\DTO\DocumentFieldType[]::class}}
     * }})
     * @var \App\DTO\array[]
     */
    public $field_types;
    /**
     * Name of the document type.
     * @DTA\Data(field="name")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $name;
    /**
     * ID for the organization this document belongs to.
     * @DTA\Data(field="orgId")
     * @DTA\Validator(name="Type", options={"type":"int"})
     * @var int
     */
    public $org_id;
    /**
     * Universally unique identifier for the document type. Can be passed in as a documentTypeUuid when creating a document for this document type.
     * @DTA\Data(field="uuid")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $uuid;
}
